gradle/artifact-settings.gradle

```diff
@@ -14,5 +13,18 @@ ext {
     mapboxArtifactScmUrl = 'scm:git@github.com:mapbox/mapbox-java.git'
     mapboxArtifactLicenseName = 'The Apache Software License, Version 2.0'
     mapboxArtifactLicenseUrl = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
-    versionName = project.hasProperty('VERSION_NAME') ? project.property('VERSION_NAME') : System.getenv('VERSION_NAME')
-}
\ No newline at end of file
+    versionName = getVersionName()
+}
+
+def getVersionName() {
+    def cmd = "git describe --tag --abbrev=0"
+    def proc = cmd.execute()
+    def version = proc.text.trim()
+    return isSnapshot() ? "${version}-SNAPSHOT" : version
```

Comment:
This makes the snapshot version name backward-looking instead of forward-looking.

If the latest tag is `v5.9.0`, the snapshots will have a `v5.9.0-SNAPSHOT` name while they should have a `v5.10.0-SNAPSHOT` name.

Reply:
yeah.. also this will make subversion snapshots 🤔 . not perfect automatic behavior.

I'll make a regex that fixes it.

Reply:
Ok i added one and tested it on all our tags in mapbox java.

```
Version 	 Snapshot
V4.8.0-alpha.1 	 v4.9.0-SNAPSHOT
v1.0.0 	 v1.1.0-SNAPSHOT
v1.1.0 	 v1.2.0-SNAPSHOT
v1.2.0 	 v1.3.0-SNAPSHOT
v1.2.1 	 v1.3.0-SNAPSHOT
v1.3.0 	 v1.4.0-SNAPSHOT
v1.3.1 	 v1.4.0-SNAPSHOT
v2.0.0 	 v2.1.0-SNAPSHOT
v2.0.0-beta.1 	 v2.1.0-SNAPSHOT
v2.0.0-beta.2 	 v2.1.0-SNAPSHOT
v2.0.0-beta.3 	 v2.1.0-SNAPSHOT
v2.1.0 	 v2.2.0-SNAPSHOT
v2.1.1 	 v2.2.0-SNAPSHOT
v2.1.2 	 v2.2.0-SNAPSHOT
v2.1.3 	 v2.2.0-SNAPSHOT
v2.2.0 	 v2.3.0-SNAPSHOT
v2.2.1 	 v2.3.0-SNAPSHOT
v2.2.10 	 v2.3.0-SNAPSHOT
v2.2.2 	 v2.3.0-SNAPSHOT
v2.2.3 	 v2.3.0-SNAPSHOT
v2.2.4 	 v2.3.0-SNAPSHOT
v2.2.5 	 v2.3.0-SNAPSHOT
v2.2.6 	 v2.3.0-SNAPSHOT
v2.2.7 	 v2.3.0-SNAPSHOT
v2.2.8 	 v2.3.0-SNAPSHOT
v2.2.9 	 v2.3.0-SNAPSHOT
v3.0.0 	 v3.1.0-SNAPSHOT
v3.0.0-beta.1 	 v3.1.0-SNAPSHOT
v3.0.0-beta.2 	 v3.1.0-SNAPSHOT
v3.0.0-beta.3 	 v3.1.0-SNAPSHOT
v3.0.0-beta.4 	 v3.1.0-SNAPSHOT
v3.0.1 	 v3.1.0-SNAPSHOT
v3.1.0 	 v3.2.0-SNAPSHOT
v3.2.0 	 v3.3.0-SNAPSHOT
v3.3.0 	 v3.4.0-SNAPSHOT
v3.4.0 	 v3.5.0-SNAPSHOT
v3.4.1 	 v3.5.0-SNAPSHOT
v4.0.0 	 v4.1.0-SNAPSHOT
v4.1.0 	 v4.2.0-SNAPSHOT
v4.1.1 	 v4.2.0-SNAPSHOT
v4.2.0 	 v4.3.0-SNAPSHOT
v4.3.0 	 v4.4.0-SNAPSHOT
v4.4.0 	 v4.5.0-SNAPSHOT
v4.5.0 	 v4.6.0-SNAPSHOT
v4.6.0 	 v4.7.0-SNAPSHOT
v4.7.0 	 v4.8.0-SNAPSHOT
v4.7.0-alpha.1 	 v4.8.0-SNAPSHOT
v4.8.0 	 v4.9.0-SNAPSHOT
v4.8.0-alpha.1 	 v4.9.0-SNAPSHOT
v4.8.0-alpha.2 	 v4.9.0-SNAPSHOT
v4.8.0-alpha.3 	 v4.9.0-SNAPSHOT
v4.9.0 	 v4.10.0-SNAPSHOT
v4.9.0-alpha.1 	 v4.10.0-SNAPSHOT
v4.9.1 	 v4.10.0-SNAPSHOT
v5.0.0 	 v5.1.0-SNAPSHOT
v5.1.0 	 v5.2.0-SNAPSHOT
v5.1.0-beta.1 	 v5.2.0-SNAPSHOT
v5.1.0-beta.2 	 v5.2.0-SNAPSHOT
v5.2.0 	 v5.3.0-SNAPSHOT
v5.2.1 	 v5.3.0-SNAPSHOT
v5.3.0 	 v5.4.0-SNAPSHOT
v5.4.0 	 v5.5.0-SNAPSHOT
v5.4.1 	 v5.5.0-SNAPSHOT
v5.5.0 	 v5.6.0-SNAPSHOT
v5.6.0 	 v5.7.0-SNAPSHOT
v5.7.0 	 v5.8.0-SNAPSHOT
v5.7.0-beta 	 v5.8.0-SNAPSHOT
v5.8.0 	 v5.9.0-SNAPSHOT
v5.8.0-beta.2 	 v5.9.0-SNAPSHOT
v5.8.0-beta.3 	 v5.9.0-SNAPSHOT
v5.8.0-beta.4 	 v5.9.0-SNAPSHOT
v5.9.0-alpha.1 	 v5.10.0-SNAPSHOT
```